app/services/idv/gpo_mail.rb

```diff
@@ -41,39 +43,26 @@ def last_not_too_recent_enabled?
       IdentityConfig.store.minimum_wait_before_another_usps_letter_in_hours != 0
     end
 
-    attr_reader :current_user
-
     def too_many_letter_requests_within_window?
       return false unless window_limit_enabled?
-
-      number_of_letter_requests_within(
-        IdentityConfig.store.max_mail_events_window_in_days.days,
-        maximum: IdentityConfig.store.max_mail_events,
-      ) >= IdentityConfig.store.max_mail_events
+      current_user.gpo_confirmation_codes.where(
+        updated_at: Range.new(
+          IdentityConfig.store.max_mail_events_window_in_days.days.ago,
+          Time.zone.now,
+        ),
+      ).count > IdentityConfig.store.max_mail_events
     end
 
     def last_letter_request_too_recent?
       return false unless last_not_too_recent_enabled?
       return false unless current_user.gpo_verification_pending_profile?
 
-      number_of_letter_requests_within(
-        IdentityConfig.store.minimum_wait_before_another_usps_letter_in_hours.hours,
-        maximum: 1,
-        for_profile: current_user.gpo_verification_pending_profile,
-      ) > 0
-    end
-
-    def number_of_letter_requests_within(time_window, maximum:, for_profile: nil)
-      profile_query_conditions = { user: current_user }
-      profile_query_conditions[:id] = for_profile.id if for_profile
-
-      GpoConfirmationCode.joins(:profile).
-        where(
-          updated_at: (time_window.ago..),
-          profile: profile_query_conditions,
-        ).
-        limit(maximum).
-        count
+      current_user.gpo_verification_pending_profile.gpo_confirmation_codes.exists?(
+        [
+          'updated_at > ?',
```

Comment:
Do we want this to be `created_at`? I think codes could be updated when we send a reminder email

Reply:
Good question! I went with the existing code. Let me poke at what happens if you update these.

Reply:
I'm actually not finding any cases in code where we update one of these, though it's possible it's being loaded indirectly. So the difference might be academic?

Reply:
I think we should go with a change to `created_at` since we could start updating these at some point in the future.